src/SmoFacade/BackupFileTools.cs

```diff
@@ -15,7 +17,9 @@ public enum BackupType
 
     public static bool IsUrl(string path)
     {
-      return Regex.IsMatch(path, @"(http(|s):\/)(\/[^\s]+)+\.([a-zA-Z]+)$");
+      return Uri.TryCreate(path, UriKind.Absolute, out var uriResult)
+             && (uriResult.Scheme == Uri.UriSchemeHttp || uriResult.Scheme == Uri.UriSchemeHttps || uriResult.IsUnc)
+             && !(path.EndsWith(@"/") || path.EndsWith(@"\"));

```

Comment:
had to make a few tweaks to get the same functionality
tests confirm that it is working